### ISSUE TITLE
spago docs failing

### ISSUE THREAD

**Issue by cakekindel** [purescript/spago#752]
- OS: Arch linux
- node `v14.16.0`
- npm `v6.14.11`
- pulp `v15.0.0`
- purs `v0.14.0`
- spago `0.19.1`

### 1. create repro project
```sh
mkdir spago-docs-bug && cd spago-docs-bug && spago init
```

### 2. reference arrays in a public item
`src/Main.hs`
```purs
module Main where

import Prelude
import Effect.Console (log)

nums :: Array Int
nums = [1,2,3]

main = log $ show nums
```

### 3. generate docs
```sh
spago docs
```

## Observed output & errors
<details>
  <summary>Click to expand</summary>

```
Documentation written to: generated-docs/html
Building the search index...
"docs.json" decoding failed failed for output/Data.Generic.Rep/docs.json: An error occurred while decoding a JSON value:
  At object key 'declarations':
  Under 'Array':
  At array index 6:
  At object key 'typeArguments':
  Under 'Array':
  At array index 0:
  Expected value of type 'Couldn't parse Kind from {"annotation":[],"tag":"TypeConstructor","contents":[["Prim"],"Symbol"]}'.
"docs.json" decoding failed failed for output/Data.Symbol/docs.json: An error occurred while decoding a JSON value:
  At object key 'declarations':
  Under 'Array':
  At array index 0:
  At object key 'arguments':
  Under 'Array':
  At array index 0:
  Expected value of type 'Couldn't parse Kind from {"annotation":[],"tag":"TypeConstructor","contents":[["Prim"],"Symbol"]}'.
"docs.json" decoding failed failed for output/Data.Unit/docs.json: An error occurred while decoding a JSON value:
  At object key 'declarations':
  Under 'Array':
  At array index 0:
  At object key 'kind':
  Expected value of type 'Couldn't parse Kind from {"annotation":[],"tag":"TypeConstructor","contents":[["Prim"],"Type"]}'.
"docs.json" decoding failed failed for output/Effect.Uncurried/docs.json: An error occurred while decoding a JSON value:
  At object key 'declarations':
  Under 'Array':
  At array index 0:
  At object key 'kind':
  Expected value of type 'Couldn't parse Kind from {"annotation":[],"tag":"TypeApp","contents":[{"annotation":[],"tag":"TypeApp","contents":[{"annotation":[],"tag":"TypeConstructor","contents":[["Prim"],"Function"]},{"annotation":[],"tag":"TypeConstructor","contents":[["Prim"],"Type"]}]},{"annotation":[],"tag":"TypeApp","contents":[{"annotation":[],"tag":"TypeApp","contents":[{"annotation":[],"tag":"TypeConstructor","contents":[["Prim"],"Function"]},{"annotation":[],"tag":"TypeConstructor","contents":[["Prim"],"Type"]}]},{"annotation":[],"tag":"TypeConstructor","contents":[["Prim"],"Type"]}]}]}'.
"docs.json" decoding failed failed for output/Effect/docs.json: An error occurred while decoding a JSON value:
  At object key 'declarations':
  Under 'Array':
  At array index 0:
  At object key 'kind':
  Expected value of type 'Couldn't parse Kind from {"annotation":[],"tag":"TypeApp","contents":[{"annotation":[],"tag":"TypeApp","contents":[{"annotation":[],"tag":"TypeConstructor","contents":[["Prim"],"Function"]},{"annotation":[],"tag":"TypeConstructor","contents":[["Prim"],"Type"]}]},{"annotation":[],"tag":"TypeConstructor","contents":[["Prim"],"Type"]}]}'.
"docs.json" decoding failed failed for output/Prim.Boolean/docs.json: An error occurred while decoding a JSON value:
  At object key 'declarations':
  Under 'Array':
  At array index 0:
  At object key 'kind':
  Expected value of type 'Couldn't parse Kind from {"annotation":[],"tag":"TypeConstructor","contents":[["Prim"],"Boolean"]}'.
"docs.json" decoding failed failed for output/Prim.Coerce/docs.json: An error occurred while decoding a JSON value:
  At object key 'declarations':
  Under 'Array':
  At array index 0:
  At object key 'arguments':
  Under 'Array':
  At array index 0:
  Expected value of type 'Couldn't parse Kind from {"annotation":[],"tag":"TypeVar","contents":"k"}'.
"docs.json" decoding failed failed for output/Prim.Ordering/docs.json: An error occurred while decoding a JSON value:
  At object key 'declarations':
  Under 'Array':
  At array index 0:
  At object key 'kind':
  Expected value of type 'Couldn't parse Kind from {"annotation":[],"tag":"TypeConstructor","contents":[["Prim"],"Type"]}'.
"docs.json" decoding failed failed for output/Prim.Row/docs.json: An error occurred while decoding a JSON value:
  At object key 'declarations':
  Under 'Array':
  At array index 0:
  At object key 'arguments':
  Under 'Array':
  At array index 0:
  Expected value of type 'Couldn't parse Kind from {"annotation":[],"tag":"TypeApp","contents":[{"annotation":[],"tag":"TypeConstructor","contents":[["Prim"],"Row"]},{"annotation":[],"tag":"TypeVar","contents":"k"}]}'.
"docs.json" decoding failed failed for output/Prim.RowList/docs.json: An error occurred while decoding a JSON value:
  At object key 'declarations':
  Under 'Array':
  At array index 0:
  At object key 'kind':
  Expected value of type 'Couldn't parse Kind from {"annotation":[],"tag":"TypeApp","contents":[{"annotation":[],"tag":"TypeApp","contents":[{"annotation":[],"tag":"TypeConstructor","contents":[["Prim"],"Function"]},{"annotation":[],"tag":"TypeConstructor","contents":[["Prim"],"Type"]}]},{"annotation":[],"tag":"TypeConstructor","contents":[["Prim"],"Type"]}]}'.
"docs.json" decoding failed failed for output/Prim.Symbol/docs.json: An error occurred while decoding a JSON value:
  At object key 'declarations':
  Under 'Array':
  At array index 0:
  At object key 'arguments':
  Under 'Array':
  At array index 0:
  Expected value of type 'Couldn't parse Kind from {"annotation":[],"tag":"TypeConstructor","contents":[["Prim"],"Symbol"]}'.
"docs.json" decoding failed failed for output/Prim.TypeError/docs.json: An error occurred while decoding a JSON value:
  At object key 'declarations':
  Under 'Array':
  At array index 0:
  At object key 'arguments':
  Under 'Array':
  At array index 0:
  Expected value of type 'Couldn't parse Kind from {"annotation":[],"tag":"TypeConstructor","contents":[["Prim","TypeError"],"Doc"]}'.
"docs.json" decoding failed failed for output/Prim/docs.json: An error occurred while decoding a JSON value:
  At object key 'declarations':
  Under 'Array':
  At array index 0:
  At object key 'kind':
  Expected value of type 'Couldn't parse Kind from {"annotation":[],"tag":"TypeApp","contents":[{"annotation":[],"tag":"TypeApp","contents":[{"annotation":[],"tag":"TypeConstructor","contents":[["Prim"],"Function"]},{"annotation":[],"tag":"TypeConstructor","contents":[["Prim"],"Type"]}]},{"annotation":[],"tag":"TypeApp","contents":[{"annotation":[],"tag":"TypeApp","contents":[{"annotation":[],"tag":"TypeConstructor","contents":[["Prim"],"Function"]},{"annotation":[],"tag":"TypeConstructor","contents":[["Prim"],"Type"]}]},{"annotation":[],"tag":"TypeConstructor","contents":[["Prim"],"Type"]}]}]}'.
"docs.json" decoding failed failed for output/Type.Proxy/docs.json: An error occurred while decoding a JSON value:
  At object key 'declarations':
  Under 'Array':
  At array index 2:
  At object key 'typeArguments':
  Under 'Array':
  At array index 0:
  Expected value of type 'Couldn't parse Kind from {"annotation":[],"tag":"TypeApp","contents":[{"annotation":[],"tag":"TypeApp","contents":[{"annotation":[],"tag":"TypeConstructor","contents":[["Prim"],"Function"]},{"annotation":[],"tag":"TypeConstructor","contents":[["Prim"],"Type"]}]},{"annotation":[],"tag":"TypeApp","contents":[{"annotation":[],"tag":"TypeApp","contents":[{"annotation":[],"tag":"TypeConstructor","contents":[["Prim"],"Function"]},{"annotation":[],"tag":"TypeConstructor","contents":[["Prim"],"Type"]}]},{"annotation":[],"tag":"TypeConstructor","contents":[["Prim"],"Type"]}]}]}'.
Indexing 54 modules from 4 packages...
Added 235 definitions and 174 type definitions from 4 packages to the search index.
```
</details>

**Comment by f-f** [comment]
It sounds like this is happening because we need to upgrade [docs-search](https://github.com/spacchetti/purescript-docs-search) to 0.14? cc @klntsky

**Comment by f-f** [comment]
[This line](https://github.com/purescript/pursuit/blob/d0d0b0962175f2dac3a97ba558bac7edb8a8b3ad/src/SearchIndex.hs#L280) seems to be the only relevant addition to Pursuit for 0.14

**Comment by klntsky** [comment]
> It sounds like this is happening because we need to upgrade [docs-search](https://github.com/spacchetti/purescript-docs-search) to 0.14? cc @klntsky 

I'm on it.

**Comment by f-f** [comment]
It might also be a good idea to add a test that verifies that `spago docs` succeeds, to make sure that `docs-search` still works when we update the compiler

**Comment by cakekindel** [comment]
you guys rock! gonna be watching that PR so i can fix similar issues in the future :smile:

**Comment by cakekindel** [comment]
By the way incase someone hits this issue before fix is merged, I was able to work around this for the time being by running
```sh
pulp docs "src/**/*.purs" ".spago/*/*/src/**/*.purs"
```
you may also be able to use `pulp docs $(spago sources)` but I haven't tested this.

**Comment by f-f** [comment]
@cakekindel also `spago docs --no-search` should work

**Comment by f-f** [comment]
Fixed in #775
